mdbx-go overflow on Arch Linux

**Describe the bug**

Running `make test` and `make benchmarks` result in the following mdbx error:

```
# github.com/torquem-ch/mdbx-go/mdbx
In file included from mdbx.c:1036:
In function ‘atomic_store64’,
    inlined from ‘mdbx_find_oldest.isra’ at mdbx.c:9154:12:
mdbx.c:4279:3: error: ‘__atomic_store_8’ writing 8 bytes into a region of size 0 overflows the destination [-Werror=stringop-overflow=]
 4279 |   atomic_store_explicit(MDBX_c11a_rw(uint64_t, p), value, mo_c11_store(order));
      |   ^~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

**To Reproduce**

Only known to occur on Arch Linux.

1. Clone the repo
2. Run `make test` or `make benchmarks`

**Expected behavior**

Tests and benchmarks run without errors.

**Desktop**

 - OS: Arch Linux
 - Version: 5.17.7-arch1-1

--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,4 @@
+export CC = clang
 .DEFAULT_GOAL 	:= help
 
 compile: ## compile:

